src/uimports.cpp

```diff
@@ -106,6 +106,34 @@ static BOOL UPDATE_IMPORTS_XX(HANDLE hProcess,
         }
     }
 
+	///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+	//Added code
+	if (inh.IMPORT_DIRECTORY.VirtualAddress!=NULL && inh.IMPORT_DIRECTORY.Size==0)
+	{
+		//Don't worry about changing the PE file, because the load information of the original PE header has been saved and will be restored. The change here is just for the following code to work normally
+		PIMAGE_IMPORT_DESCRIPTOR pImageImport=(PIMAGE_IMPORT_DESCRIPTOR)(pbModule+inh.IMPORT_DIRECTORY.VirtualAddress);
+		IMAGE_IMPORT_DESCRIPTOR ImageImport;
+
+		if(!ReadProcessMemory(hProcess,pImageImport,&ImageImport,sizeof(ImageImport),NULL))
+		{
+			_Analysis_assume_(FALSE);
+			goto finish;
+		}
+		while(ImageImport.Name)
+		{
+			inh.IMPORT_DIRECTORY.Size+=sizeof(IMAGE_IMPORT_DESCRIPTOR);
+			if(!ReadProcessMemory(hProcess,pImageImport,&ImageImport,sizeof(ImageImport),NULL))
```

Comment:
I think you could simplify this code by switching this to a do-while loop, that way the initial ReadProcessMemory doesn't need to be special cased.

Reply:
I don't quite understand your expression, and here `ImageImport.Name` is a judgment condition, so I think it is more appropriate to use a while-do loop.

Reply:
The loop maybe better:
```cpp
for (;;) {
        if (!ReadProcessMemory(hProcess, pImageImport, &ImageImport, sizeof(ImageImport), NULL)) {
            DETOUR_TRACE(("ReadProcessMemory failed: %u\n", GetLastError()));;
            goto finish;
        }
        if (ImageImport.Name) {
            ++pImageImport;
            inh.IMPORT_DIRECTORY.Size += sizeof(IMAGE_IMPORT_DESCRIPTOR);
            continue;
        }
        break;
}
```

Reply:
Updated

Reply:
Although the code becomes able to get the correct result after you update and fix it, but at the same time it is difficult to read. In my original code, only run `inh.IMPORT_DIRECTORY.Size+=sizeof(IMAGE_IMPORT_DESCRIPTOR);` under the condition of `ImageImport.Name` , and you changed this code but first executed `inh.IMPORT_DIRECTORY.Size+=sizeof(IMAGE_IMPORT_DESCRIPTOR);` and ignored the conditions of `ImageImport.Name`. It seems that it is just to make the result correct, so I think it breaks the readability and there is no need to modify it.

Reply:
why, it's easy to unstand, just calc size until the name is null

Reply:
Update again, the latest code logical is equal to your original code

Reply:
OK, now the code has been adjusted to be more in line with official recommendations.